From 7ff6515c706e881fe1388576e79d52243e290958 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Mon, 2 Jan 2012 12:27:29 +0100
Subject: [PATCH] Do not dereference NULL argument of pcre_free_study()

Back-ported from:
r836 | ph10 | 2011-12-28 18:16:11 +0100 (St, 28 pro 2011) | 2 lines
Merging all the changes from the pcre16 branch into the trunk.

Upstream bug #1186.
---
 pcre_study.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/pcre_study.c b/pcre_study.c
index cb2f23e..0a3ecbc 100644
--- a/pcre_study.c
+++ b/pcre_study.c
@@ -1377,6 +1377,8 @@ Returns:    nothing
 PCRE_EXP_DEFN void
 pcre_free_study(pcre_extra *extra)
 {
+if (extra == NULL)
+  return;
 #ifdef SUPPORT_JIT
 if ((extra->flags & PCRE_EXTRA_EXECUTABLE_JIT) != 0 &&
      extra->executable_jit != NULL)
-- 
1.7.7.5

